ksa_compliance/jinja.py

```diff
@@ -1,2 +1,81 @@
-def get_zatca_qr_for_invoice(invoice_name: str) -> str:
-    pass
+import base64
+
+import frappe
+from io import BytesIO
+from frappe.utils.data import add_to_date, get_time, getdate
+import datetime
+from base64 import b64encode, b64decode
+import pyqrcode
+
+
+@frappe.whitelist()
+def get_zatca_phase_1_qr_for_invoice(invoice_name: str) -> str:
+    values_dict = get_qr_inputs(invoice_name)
+    decoded_string = generate_decoded_string(values_dict)
+    print("decoded_string",decoded_string)
+    return generate_qrcode(decoded_string)
+
+
+def get_qr_inputs(invoice_name):
+    sales_invoice = frappe.get_doc("Sales Invoice", invoice_name)
+    seller_name = sales_invoice.company
+    if not frappe.get_value("ZATCA Phase 1 Business Settings", {"company": seller_name}):
+        return None
+    phase_1_settings = frappe.get_list("ZATCA Phase 1 Business Settings", filters={"company": seller_name}, fields=["*"])[0]
+    if phase_1_settings.status == "Disabled":
+        return None
+    seller_vat_reg_no = phase_1_settings.vat_registration_number
+    time = sales_invoice.posting_time
+    timestamp = format_date(sales_invoice.posting_date, time)
+    grand_total = sales_invoice.grand_total
+    total_vat = sales_invoice.total_taxes_and_charges
+    return {seller_name: [1], seller_vat_reg_no: [2], timestamp: [3], grand_total: [4], total_vat: [5]}
```

Comment:
This is very convoluted. I realize you didn't write this, since it's taken from `lava_custom`, but we need to refactor it into something comprehensible. Does phase 1 QR have order requirements (i.e. fields must be in a specific order?) Because this is returning a regular dictionary, which isn't guaranteed to be ordered. And it's using lists for the values, even though they all contain one item.